### ISSUE TITLE
search call freezes

### ISSUE THREAD

**Issue by denysm88** [bclavie/RAGatouille#186]
I am trying to call it on my mac with m1 using only cpu.
Index works well.
But search is just frozen. 
It prints: "Loading searcher for index wiki for the first time... This may take a few seconds" and thats all.
And according to the Activity Monitor cpu is on 0% it means it just waiting for something.

**Comment by bclavie** [comment]
Hey, could you send over the output of a `pip freeze`? It seems like you're encountering a silent issue while loading the model but unclear what it is exactly.

**Comment by denysm88** [comment]
adlfs==2024.2.0
aiohttp==3.9.3
aiosignal==1.3.1
annotated-types==0.6.0
anthropic==0.21.3
anyio==4.3.0
appnope==0.1.4
argon2-cffi==23.1.0
argon2-cffi-bindings==21.2.0
arrow==1.3.0
asgiref==3.8.1
asttokens==2.4.1
async-lru==2.0.4
attrs==23.2.0
azure-core==1.30.1
azure-datalake-store==0.0.53
azure-identity==1.15.0
azure-storage-blob==12.19.1
Babel==2.14.0
beautifulsoup4==4.12.3
bitarray==2.9.2
bleach==6.1.0
blinker==1.7.0
cachetools==5.3.3
catalogue==2.0.10
cattrs==23.1.2
certifi==2024.2.2
cffi==1.16.0
charset-normalizer==3.3.2
click==8.1.7
cohere==5.2.1
colbert-ai==0.2.19
comm==0.2.2
cryptography==42.0.5
dataclasses-json==0.6.4
datasets==2.18.0
debugpy==1.8.1
decorator==5.1.1
defusedxml==0.7.1
Deprecated==1.2.14
dill==0.3.8
dirtyjson==1.0.8
distro==1.9.0
Django==4.2.11
django-cors-headers==4.3.1
django-extensions==3.2.3
django-health-check==3.18.1
django-sslserver==0.22
djangorestframework==3.14.0
docstring_parser==0.16
docx2txt==0.8
DoubleMetaphone==1.1
drf-yasg==1.21.6
elastic-transport==8.13.0
elasticsearch==8.13.0
emoji==2.11.0
executing==2.0.1
faiss-cpu==1.8.0
fast-pytorch-kmeans==0.2.0.1
fastavro==1.9.4
fastjsonschema==2.19.1
fasttext==0.9.2
filelock==3.13.3
fire==0.5.0
Flask==3.0.2
fqdn==1.5.1
frozenlist==1.4.1
fsspec==2024.2.0
future==1.0.0
gcsfs==2024.3.1
git-python==1.0.3
gitdb==4.0.11
GitPython==3.1.43
google-api-core==2.18.0
google-auth==2.29.0
google-auth-oauthlib==1.2.0
google-cloud-core==2.4.1
google-cloud-storage==2.16.0
google-crc32c==1.5.0
google-resumable-media==2.7.0
googleapis-common-protos==1.63.0
greenlet==3.0.3
h11==0.14.0
httpcore==1.0.5
httpx==0.27.0
huggingface-hub==0.22.2
idna==3.6
inflection==0.5.1
ipykernel==6.29.4
ipython==8.23.0
ipywidgets==8.1.2
isodate==0.6.1
isoduration==20.11.0
itsdangerous==2.1.2
jedi==0.19.1
Jinja2==3.1.3
joblib==1.3.2
json5==0.9.24
jsonpatch==1.33
jsonpointer==2.4
jsonschema==4.21.1
jsonschema-specifications==2023.12.1
jupyter==1.0.0
jupyter-console==6.6.3
jupyter-events==0.10.0
jupyter-lsp==2.2.4
jupyter_client==8.6.1
jupyter_core==5.7.2
jupyter_server==2.13.0
jupyter_server_terminals==0.5.3
jupyterlab==4.1.5
jupyterlab_pygments==0.3.0
jupyterlab_server==2.25.4
jupyterlab_widgets==3.0.10
langchain==0.1.14
langchain-community==0.0.31
langchain-core==0.1.40
langchain-text-splitters==0.0.1
langdetect==1.0.9
langsmith==0.1.39
Levenshtein==0.25.0
llama-index==0.10.26
llama-index-agent-openai==0.2.2
llama-index-cli==0.1.11
llama-index-core==0.10.26
llama-index-embeddings-openai==0.1.7
llama-index-indices-managed-llama-cloud==0.1.5
llama-index-legacy==0.9.48
llama-index-llms-openai==0.1.14
llama-index-multi-modal-llms-openai==0.1.4
llama-index-program-openai==0.1.5
llama-index-question-gen-openai==0.1.3
llama-index-readers-file==0.1.13
llama-index-readers-llama-parse==0.1.4
llama-parse==0.4.0
llamaindex-py-client==0.1.16
MarkupSafe==2.1.5
marshmallow==3.21.1
matplotlib-inline==0.1.6
mistune==3.0.2
mmh3==4.1.0
mpmath==1.3.0
msal==1.28.0
msal-extensions==1.1.0
multidict==6.0.5
multiprocess==0.70.16
mypy-extensions==1.0.0
nbclient==0.10.0
nbconvert==7.16.3
nbformat==5.10.4
nest-asyncio==1.6.0
networkx==3.2.1
nicknames==0.1.8
ninja==1.11.1.1
nltk==3.8.1
notebook==7.1.2
notebook_shim==0.2.4
numpy==1.26.4
oauthlib==3.2.2
onnx==1.16.0
openai==1.16.1
orjson==3.10.0
overrides==7.7.0
packaging==23.2
pandas==2.2.1
pandocfilters==1.5.1
parsimonious==0.10.0
parso==0.8.4
pdf2image==1.17.0
pdfminer.six==20231228
pexpect==4.9.0
phonetics==1.0.5
pillow==10.3.0
platformdirs==4.2.0
portalocker==2.8.2
probableparsing==0.0.1
probablepeople==0.5.4
prometheus_client==0.20.0
prompt-toolkit==3.0.43
proto-plus==1.23.0
protobuf==4.25.3
psutil==5.9.8
psycopg2==2.9.6
ptyprocess==0.7.0
pure-eval==0.2.2
pyarrow==15.0.2
pyarrow-hotfix==0.6
pyasn1==0.6.0
pyasn1_modules==0.4.0
pybind11==2.12.0
pycparser==2.22
pydantic==2.6.4
pydantic_core==2.16.3
Pygments==2.17.2
PyJWT==2.8.0
PyMuPDF==1.24.1
PyMuPDFb==1.24.1
pynvml==11.5.0
pypdf==4.1.0
pytesseract==0.3.10
python-crfsuite==0.9.10
python-dateutil==2.9.0.post0
python-dotenv==1.0.1
python-json-logger==2.0.7
python-Levenshtein==0.25.0
pytz==2024.1
PyYAML==6.0.1
pyzmq==25.1.2
qtconsole==5.5.1
QtPy==2.4.1
RAGatouille==0.0.8.post2
rapidfuzz==3.7.0
redis==5.0.3
referencing==0.34.0
regex==2023.12.25
requests==2.31.0
requests-oauthlib==2.0.0
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rpds-py==0.18.0
rsa==4.9
safetensors==0.4.2
scikit-learn==1.4.1.post1
scipy==1.13.0
Send2Trash==1.8.3
sentence-transformers==2.6.1
setuptools==69.2.0
six==1.16.0
smmap==5.0.1
sniffio==1.3.1
soupsieve==2.5
SQLAlchemy==2.0.29
sqlparse==0.4.4
srsly==2.4.8
stack-data==0.6.3
stanza==1.8.1
striprtf==0.0.26
supar==1.1.2
svgling==0.4.0
svgwrite==1.4.3
sympy==1.12
tenacity==8.2.3
termcolor==2.4.0
terminado==0.18.1
thefuzz==0.22.1
threadpoolctl==3.4.0
tiktoken==0.6.0
tinycss2==1.2.1
tokenizers==0.15.2
toml==0.10.2
torch==2.2.2
tornado==6.4
tqdm==4.66.2
traitlets==5.14.2
transformers==4.39.3
types-python-dateutil==2.9.0.20240316
types-requests==2.31.0.20240403
typing-inspect==0.9.0
typing_extensions==4.10.0
tzdata==2024.1
ujson==5.9.0
uri-template==1.3.0
uritemplate==4.1.1
urllib3==2.2.1
voyager==2.0.6
wcwidth==0.2.13
webcolors==1.13
webencodings==0.5.1
websocket-client==1.7.0
Werkzeug==3.0.2
widgetsnbextension==4.0.10
wrapt==1.16.0
xxhash==3.4.1
yarl==1.9.4
zipfile39==0.0.8.0
